lib/register_client.rb

```diff
@@ -99,69 +107,63 @@ def get_expired_records
 
     def refresh_data
       @store.set('data') do
-        rsf = download_rsf(@register, @phase)
-        data = parse_rsf(rsf)
-        MiniCache::Data.new(data, expires_in: @config_options[:cache_duration])
+        update_cache(@register, @phase)
       end
     end
 
     private
 
     def get_data
       @store.get_or_set('data') do
-        rsf = download_rsf(@register, @phase)
-        data = parse_rsf(rsf)
-        MiniCache::Data.new(data, expires_in: @config_options[:cache_duration])
+        update_cache(@register, @phase)
       end
     end
 
-    def download_rsf(register, phase)
-      RestClient.get("https://#{register}.#{phase}.openregister.org/download-rsf")
+    def update_cache(register, phase)
+      rsf = download_rsf(register, phase, @data[:user_entry_number])
+      update_data_from_rsf(rsf, @data)
+      @data
     end
 
-    def parse_rsf(rsf)
-      items = {}
-      entries = { user: [], system: [] }
-      records = { user: {}, system: {} }
-      user_entry_number = 1
-      system_entry_number = 1
+    def download_rsf(register, phase, start_entry_number)
```

Comment:
Would it make sense to default `start_entry_number` to 0? Then the tests for this method wouldn't need to parse that argument.

Reply:
Ah yes, sorry, that should definitely be zero, will update.

Reply:
I think I misinterpreted your comment - we could make it a default value of zero but for the sake of only the tests, I don't see the need for this.